inc/namespace.php

```diff
@@ -567,44 +567,51 @@ function set_tachyon_hostname( string $tachyon_url ) : string {
 function set_s3_uploads_bucket_url_hostname( array $dirs ) : array {
 	$s3_uploads = S3_Uploads::get_instance();
 
-	$primary_host = wp_parse_url( get_main_site_url(), PHP_URL_HOST );
-	$s3_host = wp_parse_url( $s3_uploads->get_s3_url(), PHP_URL_HOST );
-	$current_host = wp_parse_url( site_url(), PHP_URL_HOST );
+	$s3_upload_dirs = wp_cache_get( 's3_upload_dirs' );
 
-	if ( ! $primary_host ) {
-		trigger_error( sprintf( 'Error parsing main site URL: %s', esc_url_raw( get_main_site_url() ) ), E_USER_WARNING );
-		return $dirs;
-	}
+	if ( ! $s3_upload_dirs ) {
```

Comment:
You can reverse this logic and return early to avoid the large if block.

Reply:
Good call, done in b7c9794.